docs/localdatetime-draft.md

```diff
@@ -0,0 +1,95 @@
+- The cases where Absolute and DateTime do the wrong thing will only produce the wrong result when used across a DST transition, so in many places the bugs will only pop up twice per year, if at all.
+- When writing the cookbook, to deal with this use case, we found that it was actually necessary a lot of the time to pass around a record consisting of `{ absolute, timeZone }`.
+  If that is so common, then we may as well have a type for it.
+- Even worse, there is a temptation to pass around a serialized string instead of such a record, which is wrong because the time zone offset may have changed when you read in the string again with `Temporal.Absolute.from()`.
+
+**API**: The API looks something like this:
+
+```typescript
+class Temporal.LocalDateTime {
+  // creation
+  constructor(absolute: Temporal.Absolute, timeZone: Temporal.TimeZone, calendar?: Temporal.Calendar);
+  static from(item: string | object, options?: object) : Temporal.LocalDateTime;
+
+  // immutable 'mutation'
+  with(localDateTimeLike: object, options?: object) : Temporal.LocalDateTime;
+  withCalendar(calendar: Temporal.Calendar) : Temporal.LocalDateTime;
+
+  // getters / field access
+  get era(): string | undefined;
+  get year(): number;
+  get month(): number;
+  get day(): number;
+  get hour(): number;
+  get minute(): number;
+  get second(): number;
+  get millisecond(): number;
+  get microsecond(): number;
+  get nanosecond(): number;
+  get timeZone(): Temporal.TimeZone;
+  get timeZoneOffsetNanoseconds(): number;
+  get timeZoneOffsetString(): string;
+  get calendar(): Temporal.Calendar;
+  get dayOfWeek(): number;
+  get dayOfYear(): number;
+  get weekOfYear(): number;
+  get hoursInDay(): number;
+  get daysInMonth(): number;
+  get daysInYear(): number;
+  get isLeapYear(): boolean;
+  getFields(): object;
+  getISOCalendarFields(): object;
+
+  // type conversion
+  toAbsolute(): Temporal.Absolute;
+  toDateTime(): Temporal.DateTime;
+  toDate(): Temporal.Date;
+  toYearMonth(): Temporal.YearMonth;
+  toMonthDay(): Temporal.MonthDay;
+  toTime(): Temporal.Time;
+
+  // comparison
+  static compare(one: Temporal.LocalDateTime, two: Temporal.LocalDateTime): number;
+  equals(other: Temporal.LocalDateTime): boolean;
+
+  // arithmetic
+  plus(duration: Temporal.Duration | object, options?: object) : Temporal.LocalDateTime;
+  minus(duration: Temporal.Duration | object, options?: object) : Temporal.LocalDateTime;
+  difference(other: Temporal.LocalDateTime, options?: object) : Temporal.Duration;
+
+  // serialization / presentation
+  toString() : string;
+  toJSON() : string;
+  toLocaleString(locales?: string | Array<string>, options?: object): string;
+}
+```
+
+(This is not a fully functional description, it's made concise for readability.
+For example, the details of the options bags are omitted.)
+
+It basically has all the API of DateTime, but with a time zone as well.
```

Comment:
I think it's important to emphasize the fact that although the API looks like DateTime+TimeZone, the data model is Absolute+TimeZone+Calendar.

Reply:
I’m not sure about this: fundamentally the “data-model” to my mind is an implementation detail. So the DateTime + TimeZone is the relevant bit.

In consequence the Absolute+TimeZone+Calendar Tuple is an implementation detail.

Reply:
I think it may be still relevant to mention, because DateTime does not unambiguously indicate a single event, and LocalDateTime does. I've added something about this.

Reply:
The public-facing API is most like DateTime+offset+TimeZone. So how about this:

> It basically has all the API of DateTime and adds a time zone, a time zone offset (for disambiguating near time zone offset transitions), and several convenience properties that depend on the time zone.